docs/design/query-caching.md

```diff
@@ -0,0 +1,224 @@
+
+For additional context see this document by Mark Nottingham titled "Caching Tutorial for Web Authors and Webmasters"
+> https://www.mnot.net/cache_docs/. 
+This document aims to explain web caching and usage of (HTTP) control caches. 
+
+Additionally, HTTP cache-control is not applicable to GraphQL due to (described later) the fact that most GraphQL requests would come via HTTP POST (not GET). More context can be found in article titled "GraphQL & Caching: The Elephant in the Room" by Marc-André Giroux.
+> https://www.apollographql.com/blog/backend/caching/graphql-caching-the-elephant-in-the-room/ 
+
+##### Overview
+
+The `cache-control`` header is defined as: 
+
+> The Cache-Control HTTP header field holds directives (instructions) — in both requests and responses — that control caching in browsers and shared caches (e.g. Proxies, CDNs). [developers.mozilla.org](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)
+
+This document's caching design applies to Data API builder at the server level and not how caching is handled at the browser or CDN level. Data API builder's behavior centers around user-defined authorization to limit the scope of data returned by the API. By definition, authenticated and subsequently authorized requests will all contain the `Authorization` HTTP header whose contents consist of the authenticated user's access token. In other words, responses are generated for specific logged-in users which per [RFC9111 - HTTP Caching](https://www.rfc-editor.org/rfc/rfc9111) states the following restriction:
+
+> A shared cache MUST NOT use a cached response to a request with an Authorization header field (Section 11.6.2 of [HTTP]) to satisfy any subsequent request unless the response contains a Cache-Control field with a response directive (Section 5.2.2) that allows it to be stored by a shared cache, and the cache conforms to the requirements of that directive for that response. [Storing Responses to Authenticated Requests](https://www.rfc-editor.org/rfc/rfc9111#name-storing-responses-to-authen)
+
+Due to how cache keys and values are created in the scope of this document, caching Data API builder responses in CDNs and browsers is not in scope. This means that `cache-control` **response** directives are not in scope. While the spec does state that the server can return `cache-control` if desired, DAB performs authentication and authorization on requests prior to accessing the in-memory cache. That same level of validation can't occur in the browser or CDN headers for security purposes. We can validate that ASP.NET core returns `private` or `no-store` as the `cache-control` header value.
+
+- Honor HTTP Header `cache-control` per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control with the following two **request** directives:
```

Comment:
So, are we saying we honor `cache-control` headers specified ONLY in ANONYMOUS request directives?

Will the following directives override whatever is configured for the respective entity in question?

Is there a directive that enforces cache for a given request even if the config has it disabled for that entity (and vice-versa)?

Reply:
I should probably just move this entire section to a different doc. I didn't remove it to avoid losing the progress made. However, this section now begins with a disclaimer that `cache-control` will come with any work related to etags when we decide to take that on. This design will not implement caching mechanisms for upstream caches like cdns/browsers, even for anonymous get requests.

Reply:
Its good to have it here, what confused me was this section: 

`Honor HTTP Header `cache-control` per https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control with the following two **request** directives:` -> which implies to me that we would honor the header for ANONYMOUS requests, but as per the disclaimer, are you saying that DAB will NOT honor this header in any case?

Reply:
> which implies to me that we would honor the header for ANONYMOUS requests, but as per the disclaimer, are you saying that DAB will NOT honor this header in any case?

Yes. With this in-memory cache, that is the case.

Reply:
updated entire section to say "NOT APPLICABLE TO IN MEMORY CACHE"